### PR TITLE
Fix Overwritten ProcessDefinitions

### LINKED ISSUE #5: Deployment "fails" silently for some diagrams
Das Deployen mancher Diagramme schlägt fehl (siehe unten). Nach kurzer Analyse bin ich darauf gekommen, dass das an der `definitionId` liegen wird (die ist in manchen der Prozesse identisch: `Definition_1`).

Die Gretchen-Frage ist nun: Wie gehen wir damit um, damit der ProcessEngine.CompatibilityProxyAdapter wie eine ProcessEngine funktioniert? Meiner Meinung nach muss da eine Weiche rein, dass bei `Definition_1` die `processModelId` anstelle der `definitionId` genommen wird.

Denn das ich 6 Prozesse erfolgreich deploye und dann "nur zwei ankommen" ist schwierig (dass `fixtures/Elements.bpmn` nicht deployt werden kann ist richtig und gewollt):

## Reproduktion

[Atlas CLI](https://github.com/atlas-engine/cli) installieren (`npm install -g @atlas-engine/cli`) und im geklonten Repo der Atlas CLI versuchen, die [Fixtures](https://github.com/atlas-engine/cli/tree/develop/fixtures) auf eine AtlasEngine zu laden:

```shell
$ atlas login http://0.0.0.0:8000 --root

Anonymous root login successful. No further steps required.

You are now logged in.


$ atlas lsp
┌─────────┐
│ (index) │
├─────────┤
└─────────┘


$ atlas deploy fixtures/*.bpmn
fixtures/Aktivierungs-E-Mails-Prozess.bpmn -> Process_1 [deployed]
fixtures/Elements.bpmn -> Elements [failed]
UnprocessableEntityError: The xml could not be parsed.
fixtures/Wartung.EMailsVerschicken.bpmn -> Wartung.EMailsVerschicken [deployed]
fixtures/Wartung.FehlerProvozieren.bpmn -> Wartung.FehlerProvozieren [deployed]
fixtures/Wartung.SekundenWarten.bpmn -> Wartung.SekundenWarten [deployed]
fixtures/Wartung.StringUmdrehen.bpmn -> Wartung.StringUmdrehen [deployed]
fixtures/wait-demo.bpmn -> wait_demo [deployed]

** (Atlas) some process models could not be deployed (see details above)


$ atlas lsp
┌─────────┬─────────────┬────────────────────┐
│ (index) │     id      │   startEventIds    │
├─────────┼─────────────┼────────────────────┤
│    0    │ 'wait_demo' │ [ 'StartEvent_1' ] │
│    1    │ 'Process_1' │ [ 'StartEvent_1' ] │
└─────────┴─────────────┴────────────────────┘
```

WDYT?

### PR DESCRIPTION
**Pull request by robpal94**
## Changes

1. This pr fixes the bug, that the deployment of some process definitions _fails_ due to a duplicated definitionId. Since the old ProcessEngine does not use the _definitionId_ at all, the BPMN Studio sets the same value for each ProcessModel.

Since the AtlasEngine internally uses the `DefinitionId`, each deployed Process which was exported with the BpmnStudio will be overwritten. 

To emulate the old behavior, where the `DefinitionId` was not used, the proxy now replaces the old definitionId with the processId, and appends a `_Definition` suffix to it. 

## Issues

Closes #5 

PR: #9

## How to test the changes

> Look at the reproduction guide in the referenced issue and confirm, that now no process silently fails to deploy.

### CODE REVIEW

File: src/management_api/proxy_services/process_model_proxy_service.ts [hunk 1]
```diff
@@ -152,6 +154,20 @@ export class ProcessModelProxyService implements ManagementAPIs.IProcessModelMan
     name: string,
     payload: ManagementApiDataModels.ProcessModels.UpdateProcessDefinitionsRequestPayload,
   ): Promise<void> {
+
+    const parser: Parser = new Parser();
```
Comment (ElRaptorus) [inline src/management_api/proxy_services/process_model_proxy_service.ts]: Wir benutzen Type Inference.

```suggestion
    const parser = new Parser();
```

File: src/management_api/proxy_services/process_model_proxy_service.ts [hunk 2]
```diff
@@ -152,6 +154,20 @@ export class ProcessModelProxyService implements ManagementAPIs.IProcessModelMan
...
+    const parsedSourceXml = await parser.parseStringPromise(payload.xml);
+
+    const processId: string = parsedSourceXml['bpmn:definitions']['bpmn:process'][0].$.id;
```
Comment (ElRaptorus) [inline src/management_api/proxy_services/process_model_proxy_service.ts]:
```suggestion
    const processId = parsedSourceXml['bpmn:definitions']['bpmn:process'][0].$.id;
```

File: src/management_api/proxy_services/process_model_proxy_service.ts [hunk 3]
```diff
@@ -152,6 +154,20 @@ export class ProcessModelProxyService implements ManagementAPIs.IProcessModelMan
...
+    const newDefinitionId = `${processId}_Definition`;
+
+    parsedSourceXml['bpmn:definitions'].$.id = newDefinitionId;
+
+    const builder: Builder = new Builder();
```
Comment (ElRaptorus) [inline src/management_api/proxy_services/process_model_proxy_service.ts]:
```suggestion
    const builder = new Builder();
```

File: src/management_api/proxy_services/process_model_proxy_service.ts [hunk 4]
```diff
@@ -152,6 +154,20 @@ export class ProcessModelProxyService implements ManagementAPIs.IProcessModelMan
...
+    const xmlWithModifiedDefinitionId: string = builder.buildObject(parsedSourceXml);
```
Comment (ElRaptorus) [inline src/management_api/proxy_services/process_model_proxy_service.ts]:
```suggestion
    const xmlWithModifiedDefinitionId = builder.buildObject(parsedSourceXml);
```

File: src/management_api/proxy_services/process_model_proxy_service.ts [hunk 5]
```diff
@@ -152,6 +154,20 @@ export class ProcessModelProxyService implements ManagementAPIs.IProcessModelMan
...
+
+    payload.xml = xmlWithModifiedDefinitionId;
```
Comment (rrrene) [inline src/management_api/proxy_services/process_model_proxy_service.ts]: Sollte dieser Block nicht besser in eine fachlich sprechende Funktion ausgelagert werden? `rewriteDefinitionIdForCompatibility` o.ä.

Comment (ElRaptorus) [reply src/management_api/proxy_services/process_model_proxy_service.ts]: Gute Idee :+1:

**Pull request merged**